pkg/gardenlet/operation/botanist/controlplane.go

```diff
@@ -53,6 +59,20 @@ func (b *Botanist) determineControllerReplicas(ctx context.Context, deploymentNa
 	return defaultReplicas, nil
 }
 
+// If the deployment is controlled by dependency-watchdog, then it has the annotation dependency-watchdog.gardener.cloud/meltdown-protection set.
+func (b *Botanist) isControlledByDependencyWatchdog(ctx context.Context, deploymentName string) (bool, error) {
+	annotations, err := kubernetesutils.GetAnnotationsForDeployment(ctx, b.SeedClientSet.Client(), b.Shoot.ControlPlaneNamespace, deploymentName)
+	if err != nil {
+		return false, err
+	}
+	if annotations != nil {
+		if _, ok := annotations[dwdapi.MeltdownProtectionActive]; ok {
+			return true, nil
+		}
+	}
+	return false, nil
+}
```

Comment:
```suggestion
	_, ok := annotations[dwdapi.MeltdownProtectionActive]
	return ok, nil
}
```

Reply:
Thanks for the optimisation. Addressed with https://github.com/gardener/gardener/pull/12272/commits/7cf655c7f2c6adbaeafe6f955ca21af3a639ef7c